blog/_posts/2022-01-17-scala-3.1.1-released.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: blog-detail
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.1.1 and forward compatibility news
+---
+
+Hello from the Scala 3 team! After a slightly longer than usual pause, we are back with a new release of the Scala compiler and exciting news about improvements in forward compatibility. The delay in the release was caused by urgent bugs that needed backporting and testing, and then the holiday season.
+
+## Improvements is Scala 3.1.1
+
+Scala 3.1.1-RC2 was promoted to a full stable release of the language. Scala 3.1.1 is the first patch version after the 3.1 minor release. It is therefore focused on bug fixes and improvements in overall correctness and stability. Although there are no new additions to the language, you can still spot some noticeable improvements:
+
+- You can use `using` clauses to introduce Scala 2 conversions (`<:<` operator).
+- Mirrors can be correctly summoned for sum types compiled by Scala 3.0.x.
+- Dual Scala 2/3 macros can be now declared in non-experimental scopes.
+- Scaladoc can correctly handle symbolic class names, such as `/` or `\`.
+- Scaladoc documentation has a new, improved UI.
+
+For a comprehensive list of fixes, take a look at the full changelog on GitHub. // TODO: link
+
+## 3.1.2-RC1 and forward compatibility improvements
+
+Scala 3 has excellent backward compatibility guarantees between the minor versions. On the other hand, after the recent release of Scala 3.1, we can see that the libraries should be really cautious with updating the compiler version, as it is forcing the bump on every user of that library. We do not want library authors to be stuck on old versions of the compiler as that would mean that they are locked out of many bugfixes, or we would need to spend enormous effort on backporting every bugfix to all past versioning lines.
+
+We are sure we want to mitigate those difficulties. We already have implemented the first and most important part of the solution and have it ready for public testing. Beginning in Scala 3.1.2-RC1, that was just released, compiler is able to generate output that can be consumed by older versions. To enable that, you need to specify a minimum version of the language using the experimental `-Yscala-release` flag.
+
+So, for example, compiling your library using Scala 3.1.2 with `-Yscala-release=3.0` flag, you make it possible for people still using Scala 3.0 to depend on your library. This has one limitation, you won't be able to use any symbol added to the standard library after 3.0. They are marked in the code and in the documentation by the internal `@Since` annotation that was also added in this version.
```

Comment:
The syntax for passing the option from the command line is `-Yscala-release 3.0` or `-Yscala-release:3.0`.  This won't work with `=` as far as I know.